js/core/options/index.js

```diff
@@ -0,0 +1,223 @@
+import createCallBack from '../utils/callbacks';
+import { isFunction, isObject, type } from '../utils/type';
+import { equalByValue } from '../utils/common';
+import { OptionManager } from './option_manager';
+import { clone } from '../utils/object';
+import { getFieldName, getParentName, convertRulesToOptions } from './utils';
+
+export class Options {
+    constructor(options, defaultOptions, optionsByReference, deprecatedOptions) {
+        this._deprecatedCallbacks = createCallBack({ syncStrategy: true });
+
+        this._default = defaultOptions;
+        this._deprecated = deprecatedOptions;
+
+        this._cachedDeprecateNames = [];
+
+        this._optionManager = new OptionManager(
+            options,
+            optionsByReference
+        );
+        this._optionManager.onRelevantNamesPrepared((options, name, value) => this._setRelevantNames(options, name, value));
+
+        this._rules = [];
+    }
+
+    get _options() {
+        return this._optionManager._options;
```

Comment:
Should we use an API to get it? I think yes because `_options` is a private property of the `this._optionManager` and we should use only public API to operate.